scripts/data_collection/ameren_scraper.py

```diff
@@ -52,174 +97,392 @@ def setup_driver() -> webdriver.Chrome:
+
+    print(f"\n{'=' * 80}")
+    print(f"Processing: {filename}")
+    print(f"{'=' * 80}")
+
+    # Check if file already exists in S3
+    exists = file_exists_in_s3(s3_client, bucket, s3_key)
+    if exists and not force and not ask_overwrite_permission(filename, s3_key, bucket):
+        print(f"Skipping: {filename}")
+        return True
+    if exists:
+        print("  Will overwrite existing file")
+
+    # Create fresh browser session for this file
     driver = setup_driver()
     try:
+        print("Creating fresh session...")
         cookies = login_and_get_cookies(driver, username, password)
-        download_all_csvs(cookies, driver, DOWNLOAD_DIR, FOLDER_URL)
+
+        # Download to local temp directory
+        local_path = download_csv(file_url, cookies, DOWNLOAD_DIR)
+        if not local_path:
+            return False
+
+        # Upload to S3
+        upload_success = upload_to_s3(s3_client, local_path, bucket, s3_key)
+
+        # Clean up local file after successful upload
+        if upload_success:
+            try:
+                local_path.unlink()
+                print("Deleted local file")
+            except Exception as e:
+                print(f"Warning: Could not delete local file: {e}")
+
+        return upload_success
+
+    finally:
+        driver.quit()
+        print("Closed browser session")
+
+
+def main():
+    """Main entry point: process each file with its own isolated browser session."""
+    parser = argparse.ArgumentParser(description="Download Ameren CSV files and upload to S3")
+    parser.add_argument(
+        "--bucket-name", default=DEFAULT_S3_BUCKET, help=f"S3 bucket name (default: {DEFAULT_S3_BUCKET})"
+    )
+    parser.add_argument("--force", action="store_true", help="Skip confirmation prompts for existing files")
+    args = parser.parse_args()
+
+    print("Enhanced Ameren CSV Downloader with S3 Integration")
+    print(f"Target S3 bucket: {args.bucket_name}\n")
+
+    # Load credentials
+    try:
+        creds = load_credentials(CREDENTIALS_FILE)
+        username, password = creds["username"], creds["password"]
+    except Exception as e:
+        print(f"ERROR: Failed to load credentials from {CREDENTIALS_FILE}: {e}")
+        return
+
+    # Initialize S3 client
+    try:
+        s3_client = boto3.client("s3")
+        s3_client.list_buckets()
+    except Exception as e:
+        print(f"ERROR: Failed to connect to S3: {e}")
+        print("Make sure AWS credentials are configured")
+        return
+
+    # Get list of CSV files with initial browser session
+    print("Scanning for CSV files...")
+    driver = setup_driver()
+    try:
+        # Login for session side-effects; cookies not used directly here
+        login_and_get_cookies(driver, username, password)
+        csv_urls = list_csv_urls_in_folder(driver, FOLDER_URL)
     finally:
         driver.quit()
 
+    if not csv_urls:
+        print("ERROR: No CSV files found")
+        return
```

Comment:
Is this idea that the script terminates at this point? If so, you should exit with an appropriate error code. (Look up unix process error codes to see why how they work and why they're important... they let any code that is calling this script handle different outcomes of the script itself.)